lib/web_of_science/process_records.rb

```diff
@@ -98,33 +98,18 @@ def create_publication(record)
       # @return [void]
       def process_links
         links = retrieve_links
-        records.each { |rec| update_links(rec, links[rec.uid]) }
+        records.each { |rec| rec.identifiers.update(links[rec.uid]) if rec.database == 'WOS' }
       rescue StandardError => err
-        message = "Author: #{author.id}, process_links failed"
-        NotificationManager.error(err, message, self)
+        NotificationManager.error(err, "Author: #{author.id}, process_links failed", self)
       end
```

Comment:
Not sure I can approve this change because if any exception arises when updating one record, it will kill the iteration for all the rest of the records.  That's why it's split out into two methods, so the method that handles the links update for one record can handle and log any exception for it without killing the iteration across records.

Reply:
Right now there is basically no validation on pub IDs, so the circumstances under which `update` might fail would be extraordinary. Previously, the same code could *still* fail for all records in the more likely spot of the `retrieve_links` remote call, so I don't think the cost of separate method and `rescue` context actually delivers the desired bulletproofing.

Reply:
I can't tell if this is intended to be a performance improvement to reduce the number of queries or a simplification/refactor?

Reply:
Mainly the latter.  I don't think it makes sense to expect local DB failures here (compared to expecting potential network failures for remote requests).  Performance improvement would be from avoiding unnecessary methods and rescue contexts (relatively small compared to reducing queries).